src/control/SConscript

```diff
@@ -66,9 +66,27 @@ def install_go_bin(denv, gosrc, gopath, libs, name, install_name):
     src = [mod_src]
     if libs is not None:
         src.extend(libs)
+    # generate a unique build id per binary for use by RPM
+    # https://fedoraproject.org/wiki/PackagingDrafts/Go#Build_ID
+    def gen_build_id():
+        from os import urandom
+        from binascii import b2a_hex
+        buildid = b2a_hex(urandom(20))
+        if isinstance(buildid, str):
+            # Python 2 is a str already, using decode() here transforms it
+            # into a 'unicode' type
+            return '0x' + buildid
+        else:
+            # Python 3 is a bytes, decode() it to a str
+            return '0x' + buildid.decode()
+    def go_ldflags():
+        return ' '.join(['-ldflags',
+                         '"-X main.daosVersion=%s' % DAOS_VERSION,
+                         '-B %s"' % gen_build_id(),
+                         '-v'])
```

Comment:
Technically, `-v` is a flag for `go install`, but I don't think you need to rework it for that.

Reply:
Ah, right.  Of course.